drift-client/src/main/java/io/airlift/drift/client/DriftMethodInvocation.java

```diff
@@ -71,6 +72,8 @@
     @GuardedBy("this")
     private final Set<A> attemptedAddresses = new LinkedHashSet<>();
     @GuardedBy("this")
+    private final Map<A, Integer> failedConnectionAttempts = new HashMap<>();
```

Comment:
I think this would be cleaner as a `Multiset`. Then you can use the simpler `count()` and `add()` methods.

Reply:
That is even better than my suggestion.

Reply:
Thanks for the suggestion. Originally i was looking for something like this, but totally forgot how is it called in Guava (was looking for something like CountingMap).